st2common/st2common/models/db/__init__.py

```diff
@@ -271,10 +272,13 @@ def get_by_ref(self, value):
     def get_by_pack(self, value):
         return self.get(pack=value, raise_exception=True)
 
-    def get(self, exclude_fields=None, *args, **kwargs):
+    def get(self, *args, **kwargs):
```

Comment:
I saw your comment on the PR about keyword args, but I'm not sure exactly what kind of issue you had in Python 2.5.

Combining keyword arguments + `**kwargs` should work fine, we use it in many places (and I really don't like `**kwargs` stuff in general since it makes code hard to read and understand what is going on, but sadly that's legacy in some places). Am I missing something?

Reply:
Python 3 lets you put `*args` before any arguments that you want to be keyword-only arguments. Like this:

```python
def asdf(arg1, *args, kwarg=None, **kwargs):
    pass
```

But the exact same code will get you a syntax error in Python 2.7 (and presumably below).

```python
>>> def asdf(arg1, *args, kwarg=None, **kwargs):
  File "<stdin>", line 1
    def asdf(arg1, *args, kwarg=None, **kwargs):
                              ^
SyntaxError: invalid syntax
```

I have not checked Python 2.5 yet. Are some StackStorm users still running it with that?

Reply:
Expanding a bit more on this change, `exclude_fields` is not a keyword-only argument in Python 2, so if I were to try to use `Q` objects, like so:

```python
ActionAlias.get(
    Q(formats__representations__match_multiple=None)
    | Q(formats__representations__match_multiple=False),
    enabled=True)
```

then within the original `.get` function, it would unpack to:

```python
exclude_fields = Q(
    QCombination(
        operation=QCombination.OR,
        children=(QNode(formats__representations__match_multiple=None),
                  QNode(formats__representations__match_multiple=False)))
args = (,)  # Empty tuple
kwargs = {}  # Empty dict
```

Now, a `QCombination` object is not a reasonable value for the `exclude_fields` variable, so code that used `exclude_fields` could explode later on.

The alternative that I have coded up makes `exclude_fields` an **implicit**, keyword-only argument in Python 2. This isn't ideal, but there's not really any reasonable way to make the `.get` function accept `*args` without confusing it with `exclude_fields`.

I briefly considered checking 

```python
if instanceof(exclude_fields, QCombination) and not args:
    args = exclude_fields
    exclude_fields = kwargs.pop('exclude_fields')
```

but that seemed like too much of a hack to me.

Only Python 3 will allow us to make `exclude_fields` an **explicit**, keyword-only argument.